language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -624,7 +626,7 @@ fn parse_term<'input>(
 fn parse_struct_name<'input>(
     tokens: &mut Lexer<'input>,
 ) -> Result<StructName, ParseError<usize, anyhow::Error>> {
-    StructName::parse(parse_name(tokens)?)
+    Ok(StructName::parse(parse_name(tokens)?)?)
```

Comment:
What is the point of unwrapping the Result and the re-wrapping it? Is this a Rust convention or something? Ditto for the change to parse_module_name.

Reply:
This is basically to force an error conversion from `anyhow::Error` to `ParseError`. The `?` operator does an `into()` call.

Reply:
Got it. That makes sense